src/tests/Microsoft.PowerFx.Interpreter.Tests/DeferredTypeTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Tests.Helpers;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    public class DeferredTypeTests
+    {
+        [Theory]
+        [InlineData("X", "X")]
+        [InlineData("$\"Test {X}\"", "s")]
+        [InlineData("!X", "b")]
+        [InlineData("Table(X)", "X")]
+        [InlineData("X.Field1", "X")]
+
+        [InlineData("X + 1", "X")] // resolve
+        [InlineData("X + \"1\"", "X")] // resolve
+        [InlineData("X + DateTime(2022, 11, 10, 0, 0, 0)", "d")]
+        [InlineData("X + Date(2022, 11, 10)", "d")]
+        [InlineData("X + Time(0, 0, 0)", "T")]
+        [InlineData("1 + X", "X")] // resolve
+        [InlineData("DateTime(2022, 11, 10, 0, 0, 0) + X", "d")]
+        [InlineData("Date(2022, 11, 10) + X", "d")]
+        [InlineData("Time(0, 0, 0) + X", "T")]
+
+        [InlineData("X * 1", "n")]
+        [InlineData("And(X, 1=1)", "b")]
+        [InlineData("X&\"test\"", "s")]
+        [InlineData("X = 1", "b")]
+        [InlineData("X = GUID(\"5cc45615-f759-4a53-b225-d3a2497f60ad\")", "b")]
+
+        [InlineData("X < 2", "b")]
+        [InlineData("X < DateTime(2022,11,10,0,0,0)", "b")]
+        [InlineData("\"The\" in X", "b")]
+        [InlineData("123 in X", "b")]
+        [InlineData("X in [1, 2, 3]", "b")]
+        [InlineData("123 in Table(X)", "b")]
+        [InlineData("IsBlank(X)", "b")]
+        [InlineData("IsError(X)", "b")]
+        [InlineData("Text(X)", "s")]
+        [InlineData("Value(X)", "n")]
+        [InlineData("Boolean(X)", "b")]
+        [InlineData("Index([1,2,3].Value, X)", "![Value:n]")]
+        [InlineData("Sum(X, 1) < 5", "b")]
+        [InlineData("Sum(X, 1, R) < 5", "b")] // All error are discarded for function calls, hence we don't get error for RecordType here.
+        [InlineData("Sum(X, T) < 5", "b")] // Since we discard all errors for function calls, Function calls are biased to non tabular overload.
+
+        [InlineData("Sum(X, X)", "n")]
+        [InlineData("X.Field1.Field1", "X")]
+        [InlineData("If(true, X)", "X")]
+        [InlineData("If(true, X, 1)", "n")]
+        [InlineData("If(true, X, X)", "X")]
+        [InlineData("ForAll([1,2,3], X)", "X")]
+        [InlineData("ForAll(ParseJSON(\"[1]\"), X)", "X")]
+        [InlineData("Abs(Table(X))", "n")]
```

Comment:
>"n" [](http://example.com/codeflow?start=37&length=3)

Can you add a comment to explain why this is n (and not X)

Reply:
Function overload resolution will always stay biased to scaler, and Abs scaler's returnType is number hence number